src/js/structs/__tests__/Node-test.js

```diff
@@ -108,18 +108,36 @@ describe("Node", function() {
   });
 
   describe("#getUsageStats", function() {
-    it("returns usage stats for given resource", function() {
-      const node = new Node({
-        resources: { cpus: 10 },
-        used_resources: { cpus: 5 }
+    describe("with resources from mesos", function() {
+      it("returns usage stats for given resource", function() {
+        const node = new Node({
+          resources: { cpus: 10 },
+          used_resources: { cpus: 5 }
+        });
+        const stats = {
+          percentage: 50,
+          total: 10,
+          value: 5
+        };
+
+        expect(node.getUsageStats("cpus")).toEqual(stats);
       });
-      const stats = {
-        percentage: 50,
-        total: 10,
-        value: 5
-      };
+    });
 
-      expect(node.getUsageStats("cpus")).toEqual(stats);
+    describe("without resources from mesos", function() {
+      it("returns empty resources and used_resources", function() {
+        const node = new Node({
+          resources: null,
```

Comment:
I think this is really misleading. At first I thought, "oh I must have misread something, it seems to check for null, too", but this test doesn't really test this. I personally would love to have two "duplicate" tests, one for `null` and one for `undefined`

Reply:
I thought about having 4 tests, 2 for each field is null or undefined. Or maybe even just two, one for both undefined and both null.

You are right those tests do not cover all the code paths.